### ISSUE TITLE
Training time of Melgan vs Hifigan

### ISSUE THREAD

**Issue by v-nhandt21** [kan-bayashi/ParallelWaveGAN#307]
Do you have any quick statistic about training time of Melgan and Hifigan with the same GPUs resource?

As the report in the paper that the model size of hifigan is really smaller than Melgan but I wonder why it take me too much time for fine tuning:

- I have check the distributed code and it works well
- I tried to increase worker for dataloader, and reduce segments size (8k compare with 16k in melgan experiment)
-  There are no more process sharing the gpus

![image](https://user-images.githubusercontent.com/36092323/140260239-8e1520b7-122d-4179-905f-c37cdbf34195.png)

In short, It takes 4iters/s for melgan and 2.49iters/s for hifigan, hic can you make me clear, are there any bottleneck or it is the feature of hifigan architecture.

**Comment by kan-bayashi** [comment]
I think the hifigan discriminator is much bigger than melgan's one, which make the training a bit slower.

**Comment by v-nhandt21** [comment]
> I think the hifigan discriminator is much bigger than melgan's one, which make the training a bit slower.

Yes, I got that, one more thing make me confuse is that when I increase the batch-size (16->32), the rate iters/second decrease. 

This seems obviously, but I think the process is totally on GPUs, everything is parallel, are there any reason the speed is decreased.

ps: my GPU is not used fully, just 13GB/32GB 

I got the same situation in the origin repo of hifigan author, but I can not explain for this.

**Comment by kan-bayashi** [comment]
> one more thing make me confuse is that when I increase the batch-size (16->32), the rate iters/second decrease.

I could not understand what you want to ask.
If you increase the batch-size, the number of operations are increased.
Even the gpu memory is not full, the number of parallel operations are limited and therefore, the speed is decreased.
(All operations can be parallel != loadable on gpu memory)
